### ISSUE TITLE
Lint/NonLocalExitFromIterator reporting false positive for a return inside a lambda inside RSpec.configure

### ISSUE THREAD

**Issue by DMA57361** [bbatsov/rubocop#1803]
We have the following in the spec helper of one of our projects, to ignore a deprecation warning that we'll keep getting until the next big version of a gem we use is released:

```
RSpec.configure do |config|
  # some configuration

  if Gem.loaded_specs["paper_trail"].version < Gem::Version.new("4.0.0")
    current_behavior = ActiveSupport::Deprecation.behavior
    ActiveSupport::Deprecation.behavior = lambda do |message, callstack|
      return if message =~ /`serialized_attributes` is deprecated without replacement/ && callstack.any? { |m| m =~ /paper_trail/ }
      Array.wrap(current_behavior).each { |behavior| behavior.call(message, callstack) }
    end
  end

  # more configuration
end
```

Version 0.30.0 of RuboCop is reporting the following:

```
spec/spec_helper.rb:36:7: W: Non-local exit from iterator, without return value. next, break, Array#find, Array#any?, etc. is preferred.
      return if message =~ /`serialized_attributes` is deprecated without replacement/ && callstack.any? { |m| m =~ /paper_trail/ }
```
There's no iterator here, so this looks like is could be a false positive.

**Comment by bbatsov** [comment]
//cc @ypresto

**Comment by ypresto** [comment]
It is false positive without fail, as return in lambda just continues from next line of lambda#call and harmless. It should be fixed somehow.

**Comment by ngan** [comment]
This is also a problem for the `->` lambda syntax as well.
